homeassistant/components/unifiled/light.py

```diff
@@ -2,48 +2,32 @@
 import logging
 
 from unifiled import unifiled
-import voluptuous as vol
 
-from homeassistant.components.light import (
-    ATTR_BRIGHTNESS,
-    PLATFORM_SCHEMA,
-    SUPPORT_BRIGHTNESS,
-    Light,
-)
-from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
-import homeassistant.helpers.config_validation as cv
+from homeassistant.components.light import ATTR_BRIGHTNESS, SUPPORT_BRIGHTNESS, Light
+from homeassistant.exceptions import PlatformNotReady
 
-_LOGGER = logging.getLogger(__name__)
+from .const import DOMAIN
 
-# Validation of the user's configuration
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_HOST): cv.string,
-        vol.Required(CONF_USERNAME): cv.string,
-        vol.Required(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_PORT, default=20443): vol.All(cv.port, cv.string),
-    }
-)
+_LOGGER = logging.getLogger(__name__)
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_entry(hass, config_entry, async_add_devices):
     """Set up the Unifi LED platform."""
 
     # Assign configuration variables.
-    # The configuration check takes care they are present.
-    host = config[CONF_HOST]
-    port = config[CONF_PORT]
-    username = config[CONF_USERNAME]
-    password = config[CONF_PASSWORD]
+    _ip = config_entry.data["host"]
+    _port = config_entry.data["port"]
+    _username = config_entry.data["username"]
+    _password = config_entry.data["password"]
 
-    api = unifiled(host, port, username=username, password=password)
+    api = unifiled(_ip, _port, username=_username, password=_password)
 
     # Verify that passed in configuration works
     if not api.getloginstate():
```

Comment:
I/O

Reply:
Moved this, I am not sure if it is done correctly though. If you could take a look.